components/server/src/server.ts

```diff
@@ -74,6 +77,21 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
     public async init(app: express.Application) {
         log.info('Initializing');
 
+        // metrics
+        app.use(async (req, res, next) => {
+            const startTime = Date.now()
+            const originalSend = res.send.bind(res);
+            res.send = (body) => {
+                const result = originalSend(body);
+                const method = req.method;
+                const route = req.route.path;
```

Comment:
@akosyakov, I read the following error quite a lot on prod. 

> TypeError: Cannot read property 'path' of undefined

Reply:
looks like unexpected requests for which we don't have route, could you file an issue?

Reply:
https://github.com/gitpod-io/gitpod/issues/3976